src/edu/psu/sweng500/team8/coreDataStructures/Cell.java

```diff
@@ -104,4 +104,41 @@ public void load(BinaryInputStream stream) throws IOException {
 			this.type = ValueType.Given;
 		}
 	}
+
+	@Override
+	public boolean equals(Object o) {

```

Comment:
The erratic behavior of action being entered more than once didn't exist before. It's happening either through cell value change listener firing the event and/or number input pad mouse event getting mixed up with cell's mouse event. I tried to debug, but I couldn't pin point the root cause of inconsistency. Instead of wasting time, I chose to fix the symptom. The duplicate check will make sure the redo/undo behave the way it's supposed to be.
This could cause a performance problem (it won't for our project, but in real project in similar behavior could) and might need to be investigated post-production or if you have some extra time, take a look at enterNumber method calls.